PHP_CodeSniffer/Barracuda/Sniffs/Commenting/FunctionCommentSniff.php

```diff
@@ -0,0 +1,555 @@
+                    $phpcsFile->addError($error, $tag, 'MissingParamName');
+                }//end if
+            } else {
+                $error = 'Missing parameter type';
+                $phpcsFile->addError($error, $tag, 'MissingParamType');
+            }//end if
+
+            $params[] = array(
+                         'tag'          => $tag,
+                         'type'         => $type,
+                         'var'          => $var,
+                         'comment'      => $comment,
+                         'commentLines' => $commentLines,
+                         'type_space'   => $typeSpace,
+                         'var_space'    => $varSpace,
+                        );
+        }//end foreach
+
+        $realParams  = $phpcsFile->getMethodParameters($stackPtr);
+        $foundParams = array();
+
+        foreach ($params as $pos => $param) {
+            // If the type is empty, the whole line is empty.
+            if ($param['type'] === '') {
+                continue;
+            }
+
+            // Check the param type value.
+            $typeNames = explode('|', $param['type']);
+            foreach ($typeNames as $typeName) {
+                $suggestedName = PHP_CodeSniffer::suggestType($typeName);
+                if ($typeName !== $suggestedName && !in_array($typeName, $this->customAllowedTypes)) {
+                    $error = 'Expected "%s" but found "%s" for parameter type';
+                    $data  = array(
+                              $suggestedName,
+                              $typeName,
+                             );
+
+                    $fix = $phpcsFile->addFixableError($error, $param['tag'], 'IncorrectParamVarName', $data);
+                    if ($fix === true) {
+                        $content  = $suggestedName;
+                        $content .= str_repeat(' ', $param['type_space']);
+                        $content .= $param['var'];
+                        $content .= str_repeat(' ', $param['var_space']);
+                        if (isset($param['commentLines'][0]) === true) {
+                            $content .= $param['commentLines'][0]['comment'];
+                        }
+
+                        $phpcsFile->fixer->replaceToken(($param['tag'] + 2), $content);
+                    }
+                } else if (count($typeNames) === 1 && !in_array($typeName, $this->customAllowedTypes)) {
+                    // Check type hint for array and custom type.
+                    $suggestedTypeHint = '';
+                    if (strpos($suggestedName, 'array') !== false || substr($suggestedName, -2) === '[]') {
+                        $suggestedTypeHint = 'array';
+                    } else if (strpos($suggestedName, 'callable') !== false) {
+                        $suggestedTypeHint = 'callable';
+                    } else if (strpos($suggestedName, 'callback') !== false) {
+                        $suggestedTypeHint = 'callable';
```

Comment:
This code seems to me like it's attempting to do what we did with `customAllowedTypes`. Would it make more sense to do the int/bool overrides like this, or even to move the `callback` thing into `customAllowedTypes`?